terratorch/datasets/sen4agrinet.py

```diff
@@ -0,0 +1,249 @@
+    435,   # 'Rapeseed'
+    438,   # 'Sunflower'
+    510,   # 'Potatoes'
+    770,   # 'Peas'
+]
+
+class Sen4AgriNet(NonGeoDataset):
+    def __init__(
+        self,
+        data_root: str,
+        bands: list[str] | None = None,
+        scenario: str = "random",
+        split: str = "train",
+        transform: A.Compose = None,
+        truncate_image: int | None = 4,
+        pad_image: int | None = 4,
+        interpolate_and_concat_bands: bool = True,  # noqa: FBT001, FBT002
+        seed: int = 42,
+    ):
+        """
+        Pytorch Dataset class to load samples from the Sen4AgriNet dataset, supporting
+        multiple scenarios for splitting the data.
+
+        Args:
+            data_root (str): Root directory of the dataset.
+            bands (list of str, optional): List of band names to load. Defaults to all available bands.
+            scenario (str): Defines the splitting scenario to use. Options are:
+                - 'random': Random split of the data.
+                - 'spatial': Split by geographical regions (Catalonia and France).
+                - 'spatio-temporal': Split by region and year (France 2019 and Catalonia 2020).
+            split (str): Specifies the dataset split. Options are 'train', 'val', or 'test'.
+            transform (albumentations.Compose, optional): Albumentations transformations to apply to the data.
+            truncate_image (int, optional): Number of timesteps to truncate the time dimension of the image.
+                If None, no truncation is applied. Default is 4.
+            pad_image (int, optional): Number of timesteps to pad the time dimension of the image.
+                If None, no padding is applied. Default is 4.
+            interpolate_and_concat_bands (bool): Whether to interpolate bands and concatenate them over time.
+            seed (int): Random seed used for data splitting.
+        """
+        self.data_root = Path(data_root) / "data"
+        self.transform = transform if transform else lambda **batch: to_tensor(batch)
+        self.scenario = scenario
+        self.seed = seed
+        self.truncate_image = truncate_image
+        self.pad_image = pad_image
+        self.interpolate_and_concat_bands = interpolate_and_concat_bands
+
+        if bands is None:
+            bands = ["B01", "B02", "B03", "B04", "B05", "B06", "B07", "B08", "B09", "B10", "B11", "B12", "B8A"]
+        self.bands = bands
+
+        self.image_files = list(self.data_root.glob("**/*.nc"))
+
+        self.train_files, self.val_files, self.test_files = self.split_data()
+
+        if split == "train":
+            self.image_files = self.train_files
+        elif split == "val":
+            self.image_files = self.val_files
+        elif split == "test":
+            self.image_files = self.test_files
+
+    def __len__(self):
+        return len(self.image_files)
+
+    def split_data(self):
+        random.seed(self.seed)
+
+        if self.scenario == "random":
```

Comment:
With the seed fixed, this should produce the same sorting every time, but I would check this to be sure, as lightning sets seeds somewhat inconsistently some times. I'm almost sure its fine, but if its not, it could lead to issues. Could I ask you to just manually confirm that indeed the same split is produced if the key is the same?

Reply:
I checked manually with the following code and they were equal:

`
        split_hash = self._get_split_hash(split)
        previous_hash = self._load_split_hash()

        if previous_hash is None:
            self._save_split_hash(split_hash)
            print("First")
        elif split_hash == previous_hash:
            print("Equals")
        else:
            print("Diff")
`